composer-base-components/map/map.tsx

```diff
@@ -0,0 +1,106 @@
+import { Map, Marker } from '@vis.gl/react-google-maps';
+import React, { useEffect, useState } from 'react';
+
+interface ComposerMapProps {
+  markers: { lat: number; lng: number; }[];
+  className: string;
+};
+
+type Coordinate = {
+  lat: number;
+  lng: number;
+};
+
+function ComposerMap({ markers, className }: ComposerMapProps) {
+  const [center, setCenter] = useState<Coordinate>(
+    {
+      lat: markers[0]?.lat ?? 36.8968908, // default center (antalya)
+      lng: markers[0]?.lng ?? 30.7133233,
+    }
+  );
+  const [restriction, setRestriction] = useState<google.maps.MapRestriction>(null);
+  const [defaultBounds, setDefaultBounds] = useState<google.maps.LatLngBoundsLiteral>(null);
+  const [zoom, setZoom] = useState<number>(null);
+
+  useEffect(() => {
+    if (center !== null)
+      setCenter(null);
+  }, [center]);
+
+  useEffect(() => {
+    if (restriction !== null)
+      setRestriction(null);
+  }, [restriction]);
+
+  useEffect(() => {
+    if (zoom !== null)
+      setZoom(null);
+  }, [zoom]);
+
+  useEffect(() => {
+    if (!markers.length) {
+      setZoom(3);
+      return;
+    }
+    if (markers.length === 1) {
+      setCenter({
+        lat: markers[0].lat,
+        lng: markers[0].lng
+      });
+      setZoom(14);
+      return;
+    }
+
+    const restrictions = getRestrictions();
+    const calculatedCenter = getCenter(restrictions);
```

Comment:
Since it is Bounds, you shouldn't need to set the center again. Check it, if not needed, remove it.